### DISCUSSION TITLE
UNRAID, NGINX PM, Could not Create Paste

### DISCUSSION THREAD

**Discussion by outrigger999** [PrivateBin/PrivateBin#1058] [Q&A] [answered]
Hello,

I need some help installing PrivateBin on Unraid. I did just the native install, no changes to the config, made the necessary changes to NGINX PM and can hit the web page. When I try to "Send" the message it gives me an error that it could not create paste. Error saving paste.

I have included what was in the log file below as well.

![privatebin_error_saving_paste](https://user-images.githubusercontent.com/6163106/217122840-bf8128fe-3d00-4e94-b4ea-616035cc4e28.png)

![privatebin_error_log](https://user-images.githubusercontent.com/6163106/217122865-865295d7-9dfe-4762-afaa-4612ac2df70d.png)

**Comment by elrido** [comment, marked as answer]
As the error message in your log states, the PHP process' user [requires permission to create the "data" folder and write to it](https://github.com/PrivateBin/PrivateBin/blob/master/INSTALL.md#file-level-permissions), if you use the default filesystem storage. If you prefer not give PHP write access to the data folder, you could consider setting up a database connection instead.

**Comment by outrigger999** [comment]
Just to help someone who may be faced with the same thing in the future, I solved this by using the, "Docker Safe New Perms," in the Tools section of Unraid. Run the permissions "fixer" and it will start working.

![image](https://user-images.githubusercontent.com/6163106/218568738-3340a5d8-42f2-4830-9a40-6660a3cb7802.png)

![image](https://user-images.githubusercontent.com/6163106/218568823-c9758d38-6488-451a-8202-cd97b0d9b8ed.png)
